### PR TITLE
illumos/dev-guide#25 #26 #27 #28

### PR DESCRIPTION
**Pull request by jblachly**
illumos/dev-guide#25 Needs glossary
illumos/dev-guide#26 Intro / guide layout: Grammar / diction
illumos/dev-guide#27 Guide layout references section that does not exist
illumos/dev-guide#28 Help section could benefit frmo copyediting

Textual changes are only suggestions; happy to hear feedback.

### CODE REVIEW

**Comment by rmustacc** [comment]
Thanks so much for putting this all together. Sorry it took me a bit to get back to you. I made three minor changes to this which if it's alright, I'll push with your change here (you're the still the author).

They are:

* Line wrap the guidelines section like everything else
* Link the ctf manual page in the glossary to the manual page
* Add a period at the end of See ctf(4). and See integrating changes.

**Comment by rmustacc** [comment]
@jblachly if you're okay with those, let me know and then I'll take care of the rest from here. Thanks again for doing this.

File: glossary [hunk 1]
```diff
@@ -0,0 +1,36 @@
+# Appendix: Glossary of Terms
+
+**CTF**
+Compact C Type Format. Assists in debugging. See ctf(4)
+
+**CW**
+Compiler wrapper. Historical version of illumos were built with Sun Studio
+(see below). In order to transition to the `gcc` compiler, the compiler
+wrapper was developed to translate compiler flags from Sun Studio formatted
```
Comment (richlowe) [inline glossary]: ... `to translate compiler flags from Sun Studio syntax to GCC syntax`

File: glossary [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+OS and Net were separate software components. Now this term refers to the
+operating system as a whole. 
+
+**RTI**
+Request to Integrate. The final approval process needed for a change to become
+part of the illumos gate. See [Integrating Changes](./integrating.html)
+
+**Sun Studio**
+A compiler suite historically required to build illumos. As of 2016, dependency
+on Sun Studio has been eliminated with the exception of the `lint` tool.
```
Comment (richlowe) [inline glossary]: Technically, the default env files still shadow with Studio.  I'd love to stop that though!

**Comment by jblachly** [comment]
Of course, your changes would be great. Be sure to get Rich's fix to the wording about CW.

**Comment by rmustacc** [comment]
Resolved in d91af5d.

**Comment by rmustacc** [comment]
Thanks for putting all this together @jblachly!

**Pull request closed without merging**